sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/python/PythonScanBuilder.scala

```diff
@@ -25,6 +26,24 @@ class PythonScanBuilder(
     ds: PythonDataSourceV2,
     shortName: String,
     outputSchema: StructType,
-    options: CaseInsensitiveStringMap) extends ScanBuilder {
+    options: CaseInsensitiveStringMap)
+    extends ScanBuilder
+    with SupportsPushDownFilters {
+  private var supported: Array[Filter] = Array.empty
+
   override def build(): Scan = new PythonScan(ds, shortName, outputSchema, options)
+
+  override def pushFilters(filters: Array[Filter]): Array[Filter] = {
+    val dataSource = ds.getOrCreateDataSourceInPython(shortName, options, Some(outputSchema))
+    val result = ds.source.pushdownFiltersInPython(dataSource, outputSchema, filters)
+    ds.setDataSourceInPython(dataSource.copy(dataSource = result.dataSource))
```

Comment:
So here we are updating the serialized data source instance to get the new states stored in the data source instance. Can we add some comments here (e.g when will the pushedFilters be invoked during analysis, and why do we need to set the data source instance here)

Reply:
Added comments explaining why we need to update the data source and when pushFilters is called